src/tigris.ts

```diff
@@ -334,6 +346,75 @@ export class Tigris {
 		return new Cache(this._config.projectName, cacheName, this.cacheClient, this._config);
 	}
 
+	public createOrUpdateIndex<T extends TigrisIndexType>(
```

Comment:
I think this layout is slightly confusing, for example, for the database we have all the logic related to the database inside the `Database` class which is a nice abstraction. For search/cache, we have all the operations here as top-level. Now, this means if I just look at the `createOrUpdateIndex` API, it may not be intuitive about the functionality that this API is offering. Should we have a similar class `Search` that wraps up these APIs and the user will first do `getSearch()` and then operate on it?

Reply:
how about `tigris.getSearchClient()`, `tigris.getCacheClient()`. Eventually if we have config/options specific to Search/Cache can respectively go in `new SearchClient(options)` and `new CacheClient(options)`. Would it be appropriate to call them `SearchClient` and `CacheClient`

Reply:
How will it look like for cases where the user has indexes on search and the user wants to run a search query on a collection?

Reply:
In case of collections with search index, will the user be having the search function on the collection object?

Reply:
that won't change as it we are planning to keep the search API in the collection as-is.

Reply:
Another question is how this tie in with `registerSchemas`. Here is what the flow looks like right now:

```
const tigrisClient = new Tigris();
await tigrisClient.registerSchemas([TodoItem]);
```

Would we pass the search model to this same function, then?

Reply:
Should we move registerSchemas inside the database class? Then it would be the same for search.

Reply:
Why not pass the search models in the same call as well? Instead of having separate function? Is there a benefit of having a separate function for search?

Reply:
with separate functions, for standalone indexes, there is no confusion about what to pass for database models.